src/Symfony/Component/Messenger/Command/ConsumeMessagesCommand.php

```diff
@@ -91,18 +93,32 @@ protected function configure(): void
      */
     protected function interact(InputInterface $input, OutputInterface $output)
     {
-        if (!$this->receiverNames || $this->receiverLocator->has($receiverName = $input->getArgument('receiver'))) {
-            return;
+        $style = null;
+
+        if ($this->receiverNames && !$this->receiverLocator->has($receiverName = $input->getArgument('receiver'))) {
+            $style = new SymfonyStyle($input, $output);
+            if (null === $receiverName) {
+                $style->block('Missing receiver argument.', null, 'error', ' ', true);
+                $input->setArgument('receiver', $style->choice('Select one of the available receivers', $this->receiverNames));
+            } elseif ($alternatives = $this->findAlternatives($receiverName, $this->receiverNames)) {
+                $style->block(sprintf('Receiver "%s" is not defined.', $receiverName), null, 'error', ' ', true);
+                if ($style->confirm(sprintf('Do you want to receive from "%s" instead? ', $alternatives[0]), false)) {
+                    $input->setArgument('receiver', $alternatives[0]);
+                }
+            }
         }
 
-        $style = new SymfonyStyle($input, $output);
-        if (null === $receiverName) {
-            $style->block('Missing receiver argument.', null, 'error', ' ', true);
-            $input->setArgument('receiver', $style->choice('Select one of the available receivers', $this->receiverNames));
-        } elseif ($alternatives = $this->findAlternatives($receiverName, $this->receiverNames)) {
-            $style->block(sprintf('Receiver "%s" is not defined.', $receiverName), null, 'error', ' ', true);
-            if ($style->confirm(sprintf('Do you want to receive from "%s" instead? ', $alternatives[0]), false)) {
-                $input->setArgument('receiver', $alternatives[0]);
+
+        if ($this->busNames && !$this->busLocator->has($busName = $input->getOption('bus'))) {
+            if (!$style) {
+                $style = new SymfonyStyle($input, $output);
+            }
+
+            if ($alternatives = $this->findAlternatives($busName, $this->receiverNames)) {
+                $style->block(sprintf('Bus "%s" is not defined.', $receiverName), null, 'error', ' ', true);
+                if ($style->confirm(sprintf('Do you want to dispatch to "%s" instead? ', $alternatives[0]), false)) {
+                    $input->setArgument('receiver', $alternatives[0]);
```

Comment:
this should set the bus option no? Also it's at least weird to pick the first one instead of presenting a choice question here.